File: cpp/141_LinkedListCycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        auto p = head;
        auto q = head;
        while( q != NULL && q->next != NULL){
            p = p->next;
            q = q->next->next;
            if(p == q) return true;
        }
        return false;
    }
};

File: js/141_LinkedListCycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    var p = head, q = head;
    while(q !== null && q.next !== null){
        p = p.next;
        q = q.next.next;
        if(p === q) return true;
    }
    return false;
};